Source/CronosPlayUnreal/Private/PlayCppSdkActor.cpp

```diff
@@ -23,36 +23,128 @@ using namespace com::crypto::game_sdk;
 
 ::com::crypto::game_sdk::WalletconnectClient *APlayCppSdkActor::_coreClient =
     NULL;
-APlayCppSdkActor *APlayCppSdkActor::_sdk = NULL;
 
 class UserWalletConnectCallback : public WalletConnectCallback {
+  private:
+    APlayCppSdkActor *PlayCppSdkPtr;
```

Comment:
acessing PlayCppSdkPtr from raw thread is unsafe.
```
#include "Async/Async.h"

UObject* MyUnrealObject; // UObject pointer you want to access

// Your raw thread function
void MyRawThreadFunction()
{
    // ... your code here ...

    // Use AsyncTask to execute code on the game thread
    AsyncTask(ENamedThreads::GameThread, [&]()
    {
        if (MyUnrealObject != nullptr)
        {
            // Access the UObject safely on the game thread
            // ... your code here ...
        }
    });

    // ... more code ...
}

```


and store raw pointer as weakpointer, and check `isvalid` before hand
```
TWeakObjectPtr<UClass> MyWeakClass;

// Check if the UObject is still valid before using it
if (MyWeakClass.IsValid())
{
    // Use the UObject safely
}
```

Reply:
so only game thread can access any pointer safely.

Reply:
Yes, it is not 100% fix the pointer sharing issue. I've marked it TODO:
https://github.com/cronos-labs/play-unreal-plugin/pull/308/files#diff-097b9056119eecdfb8e1f150f586de6b3ed6ef617853a734a6f0d94006274fa1R516-R517

It is unsafe but worked at this moment. Much better than the original _sdk global pointer.